builtin/game/item.lua

```diff
@@ -360,13 +360,12 @@ end
 function core.item_drop(itemstack, dropper, pos)
 	local dropper_is_player = dropper and dropper:is_player()
 	local p = table.copy(pos)
-	local cnt = itemstack:get_count()
 	if dropper_is_player then
 		p.y = p.y + 1.2
 	end
-	local item = itemstack:take_item(cnt)
-	local obj = core.add_item(p, item)
+	local obj = core.add_item(p, ItemStack(itemstack))
```

Comment:
A note to reviewers: The ItemStack was already copied before. Now thus happens more explicitly. Even though `ModApiEnv::l_add_item` does not modify `itemstack`, we should still make sure that mods cannot tamper it by accident.